### PR TITLE
Implement #1949 Background Image Align (as one setting)

### LINKED ISSUE #1949: Feature Request: Background image alignment
# Summary of the new feature/enhancement

When using the `backgroundImage` profile setting, the image is anchored to the center of the `TermControl`. Background images that have a primary point of interest that is *not* the center are not aesthetically pleasing when either... a large part of this focus is cut off via `uniformToFill`, or the image is centered via `uniform` or `none` and there is a abrupt end to the image before the border of the terminal control.

### Image Example

Here is a background designed for anchoring to the bottom right. If we rely on `uniformToFill`, then we run the risk of the image's focus getting completely cut off. If we use `none` or `uniform`, then the image will display in the middle and abruptly end on the right while the console display has more room. Alignment anchoring would allow us to display the background image as it is intended.

Preview Figure A (Left) and Figure B (Right)

*Figure A (Left), shows an image making use of alignment. Figure B (Right), shows the outcome of using the same background without alignment.*

# Proposed technical implementation details (optional)

Changes would require two additional profile settings shown below, (with center being the default for both).

```json
"backgroundImageHorizontalAlignment": "center" | "left" | "right",
"backgroundImageVerticalAlignment":   "center" | "top"  | "bottom",
```

These settings will be needed only once, in this location:

https://github.com/microsoft/terminal/blob/02e8389518c7027fa20a99b4642f29f2b6b4ef7a/src/cascadia/TerminalControl/TermControl.cpp#L273-L276

Here, `HorizontalAlignment::Center` and `VerticalAlignment::Center` will be replaced with their respective parsed settings of `backgroundImageHorizontalAlignment` and `backgroundImageVerticalAlignment`.

These two settings will need constant definitions for of their JSON/serialized values, and `Parse`/`Serialize` methods for the enums (in `Profile.cpp`).

## Alternative Setting

One enum setting that combines the flags of `HorizontalAlignment` and `VerticalAlignment` could be used. The setting and values would be as follows:

```json
"backgroundImageAlignment": "none" | "left" | "top" | "right" | "bottom"
                 | "topLeft" | "topRight" | "bottomLeft" | "bottomRight"
```

*A value of `none`, or missing axis alignment, implies that the default action of centering is used. Alternatively `none` could be labeled `center` to make this clearer.*

There are pros and cons to this alternative. The pros being the setting name is shorter, and can be fully modified from a single line in `profiles.json`. The longer enum value names are still relatively simple and easy to remember. The cons being we are not referring to a real existing enum anymore, but a combination of enums. We also require more enum definitions and handling cases than if we made each setting separate. Lastly the implied *centering* with the values of `none`, `left`, `top`, `right`, and `bottom` are not as user-friendly.

### PR DESCRIPTION
**Pull request by trigger-segfault**
## Summary of the Pull Request

Implement #1949: `profiles.json` has been given a setting for background images labeled as shown below, with the following valid values. This allows background images to be anchored to different corners/sides of the console to fit that background image's intended use-case and focus.

```json
"backgroundImageAlignment": "center" | "left" | "top" | "right" | "bottom"
                   | "topLeft" | "topRight" | "bottomLeft" | "bottomRight"
```

When `left`, `top`, `right`, or `bottom` is specified, the other axis is implied to be centered, as that is the default behavior. I went in favor of labeling the default value as `center` instead of `none` as "none" does not clearly list the alignment behavior.

### Alternative Setting

The branch [dev/trigger/background-image-align-two-settings](https://github.com/trigger-segfault/terminal/tree/dev/trigger/background-image-align-two-settings) implements the exact same functionality, but by splitting the `profiles.json` setting into separate horizontal and vertical settings. I am happy to submit it as a PR if the two-setting approach is preferred.

 
## References

There seems to be no other issues or pull requests relevant to background image alignment.

## PR Checklist
* [X] Closes #1949
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [X] Tests ~~added~~/passed
* [X] Requires documentation to be updated <sup>([SettingsSchema.md](https://github.com/microsoft/terminal/blob/master/doc/cascadia/SettingsSchema.md), [UsingJsonSettings.md](https://github.com/microsoft/terminal/blob/master/doc/user-docs/UsingJsonSettings.md) (optional))</sup>
* [X] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #1949

## Detailed Description of the Pull Request / Additional comments

The `Profile` class defines methods for parsing and serializing the combined `HorizontalAlignment` and `VerticalAlignment` enums to and from JSON. An `std::tuple<HorizontalAlignment, VerticalAlignment>` field `_backgroundImageAlignment` has been added  for storing the profile setting in one place, as it logically makes sense that both alignments to be in use together.

`TerminalSettings` and `IControlSettings` have been given `BackgroundImageHorizontalAlignment` and `BackgroundImageVerticalAlignment` properties. The combination of `std::tuple<HorizontalAlignment, VerticalAlignment>` is gone once we leave the `Profile` class. Combining the two enums is only relevant to how they are stored in `profiles.json`. `TerminalSettings` defines the default values of both alignment properties as `Center`.

The `TermControl` class changes the image setup in `TermControl::_InitializeBackgroundBrush()` to assign the `_bgImageLayer`'s (background image control's) alignments to that of the specified alignment settings. The alignments assignments have also been moved from within the image source assignment block, to the bottom of the *additional* images settings definitions (`Stretch` and `Opacity`).

### Additional Fixes

The line below for the constant definition has a lowercase `i` at the beginning of the word *"image"*. The `i` has been capitalized to follow proper UpperCamelCase. The two references to this `BackgroundImageStretchModeKey` constant have been updated to reflect the new name.

https://github.com/microsoft/terminal/blob/fad7638bb39d5c5107431605a1adf50010f8646a/src/cascadia/TerminalApp/Profile.cpp#L43

*(This change was performed in its own commit, so it can be reverted if needed.)*

## Validation Steps Performed

* [X] Test and observe all 9 possible alignment values, invalid input, and not specifying the setting at all.
  * [X] For each image stretch mode.
  * [X] For when a background image is defined or not.
  * [X] Confirm alignment names match up.
* [X] For each alignment value, confirm they are re-serialized correctly when opening Windows Terminal.
* [X] Confirm default setting results in a centered background image.
* [X] Test and observe if alignment settings are properly applied when first opening the Windows Terminal or opening a new profile tab.

## Preview Example Alignments

Below are examples of the alignments `bottom` (Left) and `bottomRight` (Right) in use.

<p align="center">Preview Figure A (Left) and Figure B (Right)</p>

*Figure A (Left) shows the use of `bottom` alignment with `uniformToFill` stretch mode. The waves will always be at the bottom and stay centered horizontally. Figure B (Right) shows the use of `bottomRight` alignment with `none` stretch mode. The image will always remain in the bottom right corner and remain the same size.*

### CODE REVIEW

**Comment by trigger-segfault** [comment]
Of course. Wasn't sure if documentation was delegated to other parties or not. I'll also add entries for the original 3 `backgroundImage` settings.

**Comment by trigger-segfault** [comment]
4 new entries have been added to the [`SettingsSchema.md`](https://github.com/microsoft/terminal/blob/master/doc/cascadia/SettingsSchema.md) documentation.

In addition, one background image-related setting capitalization typo (which results in an invalid value) has been fixed in the [`UsingJsonSettings.md`](https://github.com/microsoft/terminal/blob/master/doc/user-docs/UsingJsonSettings.md) file.

File: doc/cascadia/SettingsSchema.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ Properties listed below are specific to each unique profile.
 | `startingDirectory` | _Required_ | String | `%USERPROFILE%` | The directory the shell starts in when it is loaded. |
 | `useAcrylic` | _Required_ | Boolean | `false` | When set to `true`, the window will have an acrylic background. When set to `false`, the window will have a plain, untextured background. |
 | `background` | Optional | String | | Sets the background color of the profile. Overrides `background` set in color scheme if `colorscheme` is set. Uses hex color format: `"#rrggbb"`. |
+| `backgroundImage` | Optional | String | | Sets the file location of the Image to draw over the window background. |
+| `backgroundImageAlignment` | Optional | String | `center` | Sets how the background image aligns to the boundaries of the window. Possible values: `"center"`, `"left"`, `"top"`, `"right"`, `"bottom"`, `"topLeft"`, `"topRight"`, `"bottomLeft"`, `"bottomRight"` |
```
Comment (miniksa) [inline doc/cascadia/SettingsSchema.md]: Can you list out what the default behavior will be if you don't set the alignment/opacity/stretch given they're optional?

Comment (zadjii-msft) [reply doc/cascadia/SettingsSchema.md]: I believe that's what column 4 is for, but I'm not _sure_

Comment (trigger-segfault) [reply doc/cascadia/SettingsSchema.md]: Correct, column 4 ( **Default** ) lists the default settings of `center`, `1.0`, and `uniformToFill` respectively. Default values are listed for other optional settings in the **Globals** category as well.

**Pull request merged**